File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/impl/BerryFlavorServiceImpl.java
package io.github.lishangbu.avalon.dataset.service.impl;

import io.github.lishangbu.avalon.dataset.entity.BerryFlavor;
import io.github.lishangbu.avalon.dataset.entity.BerryFlavor_;
import io.github.lishangbu.avalon.dataset.repository.BerryFlavorRepository;
import io.github.lishangbu.avalon.dataset.service.BerryFlavorService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Example;
import org.springframework.data.domain.ExampleMatcher;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

/// 树果风味服务实现。
@Service
@RequiredArgsConstructor
public class BerryFlavorServiceImpl implements BerryFlavorService {

    private final BerryFlavorRepository berryFlavorRepository;

    @Override
    public Page<BerryFlavor> getPageByCondition(BerryFlavor berryFlavor, Pageable pageable) {
        return berryFlavorRepository.findAll(
                Example.of(
                        berryFlavor,
                        ExampleMatcher.matching()
                                .withIgnoreNullValues()
                                .withMatcher(
                                        BerryFlavor_.NAME,
                                        ExampleMatcher.GenericPropertyMatchers.contains())
                                .withMatcher(
                                        BerryFlavor_.INTERNAL_NAME,
                                        ExampleMatcher.GenericPropertyMatchers.contains())),
                pageable);
    }

    @Override
    public BerryFlavor save(BerryFlavor berryFlavor) {
        return berryFlavorRepository.save(berryFlavor);
    }

    @Override
    public BerryFlavor update(BerryFlavor berryFlavor) {
        return berryFlavorRepository.save(berryFlavor);
    }

    @Override
    public void removeById(Long id) {
        berryFlavorRepository.deleteById(id);
    }
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/impl/BerryFlavorServiceImpl.kt
package io.github.lishangbu.avalon.dataset.service.impl

import io.github.lishangbu.avalon.dataset.entity.dto.BerryFlavorSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.BerryFlavorView
import io.github.lishangbu.avalon.dataset.entity.dto.SaveBerryFlavorInput
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateBerryFlavorInput
import io.github.lishangbu.avalon.dataset.repository.BerryFlavorRepository
import io.github.lishangbu.avalon.dataset.service.BerryFlavorService
import org.babyfish.jimmer.sql.ast.mutation.SaveMode
import org.springframework.stereotype.Service

/** 树果风味服务实现 */
@Service
class BerryFlavorServiceImpl(
    /** 树果风味仓储 */
    private val berryFlavorRepository: BerryFlavorRepository,
) : BerryFlavorService {
    /** 保存树果风味 */
    override fun save(command: SaveBerryFlavorInput): BerryFlavorView = BerryFlavorView(berryFlavorRepository.save(command.toEntity(), SaveMode.INSERT_ONLY))

    /** 更新树果风味 */
    override fun update(command: UpdateBerryFlavorInput): BerryFlavorView = BerryFlavorView(berryFlavorRepository.save(command.toEntity(), SaveMode.UPSERT))

    /** 按 ID 删除树果风味 */
    override fun removeById(id: Long) {
        berryFlavorRepository.deleteById(id)
    }

    /** 按条件查询树果风味列表 */
    override fun listByCondition(
        specification: BerryFlavorSpecification,
    ): List<BerryFlavorView> = berryFlavorRepository.findAll(specification).map(::BerryFlavorView)
}
